src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -57,6 +58,64 @@ public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AppTokenPro
             }
         }
 
+        public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AcquireTokenForManagedIdentityParameters parameters, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                _requestContext.Logger.Error(TimeoutError);
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            // Convert the scopes to a resource string.
+            string resource = parameters.Resource;
+
+            ManagedIdentityRequest request = CreateRequest(resource);
+
+            try
+            {
+                HttpResponse response =
+            request.Method == HttpMethod.Get ?
+            await _requestContext.ServiceBundle.HttpManager.SendGetForceResponseAsync(request.ComputeUri(), request.Headers, _requestContext.Logger, cancellationToken: cancellationToken).ConfigureAwait(false) :
+            await _requestContext.ServiceBundle.HttpManager.SendPostForceResponseAsync(request.ComputeUri(), request.Headers, request.BodyParameters, _requestContext.Logger, cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                return await HandleResponseAsync(parameters, response, cancellationToken).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                _requestContext.Logger.Error(TimeoutError);
+                throw;
+            }
+        }
+
+        protected virtual Task<ManagedIdentityResponse> HandleResponseAsync(
```

Comment:
Seems like there is duplicate code for the `HandleResponseAsync` overloads. The `parameters` parameter isnt used so it looks like you only need one function without the `parameter`.

Reply:
Good catch. Missed to remove unused code while cleanup.